search/search-client-connect/src/main/java/io/camunda/search/connect/plugin/PluginRepository.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.connect.plugin;
+
+import io.camunda.plugin.search.header.DatabaseCustomHeaderSupplier;
+import io.camunda.zeebe.util.ReflectUtil;
+import io.camunda.zeebe.util.jar.ExternalJarLoadException;
+import io.camunda.zeebe.util.jar.ExternalJarRepository;
+import io.camunda.zeebe.util.jar.ThreadContextUtil;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Optional;
+import java.util.SequencedCollection;
+import java.util.SequencedMap;
+import java.util.stream.Stream;
+import org.agrona.LangUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Loads and holds references to the configured plugins. */
+public final class PluginRepository implements AutoCloseable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PluginRepository.class);
+
+  private final ExternalJarRepository jarRepository;
+  private final SequencedMap<String, Class<? extends DatabaseCustomHeaderSupplier>> plugins;
+  private final Path basePath;
+
+  public PluginRepository() {
+    this(
+        new LinkedHashMap<>(),
+        new ExternalJarRepository(),
+        Paths.get(Optional.ofNullable(System.getProperty("basedir")).orElse(".")));
+  }
+
+  PluginRepository(
+      final SequencedMap<String, Class<? extends DatabaseCustomHeaderSupplier>> plugins,
```

Comment:
:x: I think I made a mistake, and this should be the concrete type, `LinkedHashMap`. Here and for the field. The reason for this is `LinkedHashMap` defines the ordering of the map as the order in which keys were inserted, which matches with the order in which the plugins will be configured. Using the interface type, `SequencedMap`, we could technically pass a different implementation with a different ordering.

So we should ensure the maps used here are _always_ `LinkedHashMap` (unless there's an interface type I'm unaware of which guarantees the above semantics).

Reply:
Good point, as long as nobody decides they want to implement their own version of `LinkedHashMap` 🙃 with whatever order.
I'll change that in the next revision.